### PR TITLE
feat: implement select tokens page

### LINKED ISSUE #4: Implement select tokens page
Implement a pixel perfect select tokens page from the [FIGMA](https://www.figma.com/design/JW1DKwDBVJq97whSwlWDLS/Auto-swapper?node-id=282-1715&node-type=frame&t=TD5fHZerbCxY8HMK-0)

### PR DESCRIPTION
**Pull request by Akshola00**
# **Pull Request Template**

## **Fixes/Implements**  
**Closes #4**  

---

## **Description**  
Implemented the Select Tokens page based on the provided Figma design.  

### **Key Changes**:  
- Developed a UI for token selection following the Figma mockups.  
- Included dynamic rendering for token data (e.g., images, names, and symbols).  
- Added reusable components for rendering token cards.  
- Styled components to match the design spec.  

---

## **Related Issue(s)**  
Closes #4  

---

## **Checklist**  

- [ x] Read the [contributing docs](../CONTRIBUTING.md).  
- [x] Verified this is not a duplicate of any existing pull request.  
- [x] Performed a self-review of the code.  
- [x] Rebased/merged the latest commit from the target branch.  
- [x] Linked the issues which this PR resolves.  
- [x] Documented changes where necessary.  

---

### CODE REVIEW

**Comment by Oshioke-Salaki** [comment]
Gm, your build is failing. please fix it

**Comment by Akshola00** [comment]
fixed the build

**Comment by Oshioke-Salaki** [comment]
sync your fork and update your pr. The layout background on the layout has been changed.

**Comment by Oshioke-Salaki** [comment]
Please provide a way for me to preview your work. By creating a `select-token` route

**Comment by Akshola00** [comment]
@Oshioke-Salaki pls review

File: app/components/select-tokens.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            <SearchIcon />
+          </div>
+          <br />
+          <div className="flex  w-[318px] sm:w-[704px] flex-col">
+            <div className="flex  items-center  gap-2">
+              <TimeIcon />
+              <p>Selected tokens</p>
+            </div>
+            <div className="grid mt-4 grid-cols-2 sm:grid-cols-3 gap-1 sm:gap-4 ">
+              {coins.map((coin) => (
```
Comment (Oshioke-Salaki) [inline app/components/select-tokens.tsx:82]: One should be able to select and unselect tokens. The selected tokens should be stored in state.

File: app/components/select-tokens.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        <p className="text-center font-normal text-[14px] sm:text-[16px]  leading-[22px] w-[318px] sm:w-[620px] ">
+          You can select multiple tokens to auto-swap from here. You can select
+          how many percent of that token should be auto-swapped whenever you get
+          funded.
+        </p>
+        <br />
+
+        <form className="mx-auto flex items-center justify-center flex-col">
+          <div className="relative bg-[#100827] px-4 py-2 flex items-center justify-between w-[318px] sm:w-[704px] rounded-full">
+            <input
```
Comment (Oshioke-Salaki) [inline app/components/select-tokens.tsx:67]: The search should be functional

**Review by Oshioke-Salaki — CHANGES_REQUESTED** [review]

**Comment by Oshioke-Salaki** [comment]
Any update on this @Akshola00 ?? I need an update on this before the end of today.

**Comment by Akshola00** [comment]
@Oshioke-Salaki yes 
Will push changes before the end of today

File: app/components/select-tokens.tsx (hunk 2 above)
Comment (Akshola00) [reply app/components/select-tokens.tsx:84]: @Oshioke-Salaki the search is functional now

File: app/components/select-tokens.tsx (hunk 1 above)
Comment (Akshola00) [reply app/components/select-tokens.tsx]: @Oshioke-Salaki  the tokens are selectable and it is being stored in a state

**Comment by Akshola00** [comment]
@Oshioke-Salaki pls review

File: app/select-tokens/page.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import Selecttokens from "../components/select-tokens";
+
+const page = () => {
+  return (
+    <div className="bg-main-bg bg-cover bg-no-repeat">
```
Comment (Oshioke-Salaki) [inline app/select-tokens/page.tsx:6]: Please make sure that this takes up the full height of all devices.

File: app/components/select-tokens.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+          className="w-[318px] sm:w-[704px] text-white bg-[#100827] h-[60px] rounded-[32px]"
+          disabled={selectedCoins.length === 0}
+        >
+          Next
+        </button>
+      </div>
+    </div>
+  );
+};
+export const CloseButton = () => {
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply app/components/select-tokens.tsx:147]: done

File: app/components/select-tokens.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
+"use client";
+import React, { useState } from "react";
+import { Open_Sans } from "next/font/google";
+import btc from "../../public/coin-logos/btc-logo.png";
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply app/components/select-tokens.tsx:4]: done, gotten the svg version of the images, replaced and used it here

File: app/components/select-tokens.tsx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+import eth from "../../public/coin-logos/eth-logo.png";
+import strk from "../../public/coin-logos/strk-logo.png";
+import alt from "../../assets/alt.png";
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply app/components/select-tokens.tsx:7]: done

File: assets/backgroundtwo.svg
[the remark this answers is not in the document]
Comment (Akshola00) [reply assets/backgroundtwo.svg:1]: done

File: package-lock.json
[the remark this answers is not in the document]
Comment (Akshola00) [reply package-lock.json:1]: done

File: app/components/select-tokens.tsx [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+    setSelectedCoins((prev) =>
+      prev.includes(coinKey)
+        ? prev.filter((key) => key !== coinKey)
+        : [...prev, coinKey]
+    );
+  };
+
+  return (
+    <div className=" w-full  pt-[180px] items-center justify-center flex">
+      <div
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply app/components/select-tokens.tsx:63]: done

File: app/components/select-tokens.tsx [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+const openSans = Open_Sans({ subsets: ["latin"] });
+
+type Coin = {
+  key: string;
+  imgLink: StaticImageData | string;
+  coinName: string;
+  coinSymbol: string;
+};
+
+const Selecttokens = () => {
```
[the remark this answers is not in the document]
Comment (Akshola00) [reply app/components/select-tokens.tsx:20]: done 
made it fully responsive

**Comment by Akshola00** [comment]
@Oshioke-Salaki pls review

File: package.json
Comment (Oshioke-Salaki) [inline package.json:1]: why did you delete this? Please revert.

**Review by Oshioke-Salaki — CHANGES_REQUESTED** [review]

File: package.json
Comment (Akshola00) [reply package.json:1]: I've added it back
Pls review again

**Comment by Oshioke-Salaki** [comment]
looks good to me 🚀
